common/src/keyword.rs

```diff
@@ -38,52 +40,57 @@ pub enum Primitive {
   None,
 }
 
-impl Keyword {
-  pub fn from(op: &str) -> Option<Keyword> {
-    match op {
-      "import" => Some(Keyword::Import),
-      "as" => Some(Keyword::As),
-      "from" => Some(Keyword::From),
-      "export" => Some(Keyword::Export),
-      "fun" => Some(Keyword::Fun),
-      "return" => Some(Keyword::Return),
-      "if" => Some(Keyword::If),
-      "else" => Some(Keyword::Else),
-      "while" => Some(Keyword::While),
-      "break" => Some(Keyword::Break),
-      "continue" => Some(Keyword::Continue),
-      "var" => Some(Keyword::Var),
-      "val" => Some(Keyword::Val),
+impl TryFrom<&str> for Keyword {
+  type Error = ();
+
+  fn try_from(keyword: &str) -> Result<Self, Self::Error> {
+    match keyword {
+      "import" => Ok(Keyword::Import),
+      "as" => Ok(Keyword::As),
+      "from" => Ok(Keyword::From),
+      "export" => Ok(Keyword::Export),
+      "inline" => Ok(Keyword::Inline),
+      "fn" => Ok(Keyword::Fn),
+      "return" => Ok(Keyword::Return),
+      "if" => Ok(Keyword::If),
+      "else" => Ok(Keyword::Else),
+      "while" => Ok(Keyword::While),
+      "break" => Ok(Keyword::Break),
+      "continue" => Ok(Keyword::Continue),
+      "var" => Ok(Keyword::Var),
+      "val" => Ok(Keyword::Val),
 
-      "none" => Some(Keyword::Primitive(Primitive::None)),
-      "bool" => Some(Keyword::Primitive(Primitive::Bool)),
-      "char" => Some(Keyword::Primitive(Primitive::Char)),
-      "str" => Some(Keyword::Primitive(Primitive::Str)),
-      "i32" => Some(Keyword::Primitive(Primitive::I32)),
-      "i64" => Some(Keyword::Primitive(Primitive::I64)),
-      "u32" => Some(Keyword::Primitive(Primitive::U32)),
-      "u64" => Some(Keyword::Primitive(Primitive::U64)),
-      "f32" => Some(Keyword::Primitive(Primitive::F32)),
-      "f64" => Some(Keyword::Primitive(Primitive::F64)),
+      "none" => Ok(Keyword::Primitive(Primitive::None)),
+      "bool" => Ok(Keyword::Primitive(Primitive::Bool)),
+      "char" => Ok(Keyword::Primitive(Primitive::Char)),
+      "str" => Ok(Keyword::Primitive(Primitive::Str)),
+      "i32" => Ok(Keyword::Primitive(Primitive::I32)),
+      "i64" => Ok(Keyword::Primitive(Primitive::I64)),
+      "u32" => Ok(Keyword::Primitive(Primitive::U32)),
+      "u64" => Ok(Keyword::Primitive(Primitive::U64)),
```

Comment:
`u64` is not possible on wasm. `u32` is because it is smaller than `i64`

Reply:
Actually `u64` is possible because of the use of twos-compliment and separate `_u` opcodes for the operations not possible (div and rem i think)